File: FaturamentoService/Models/NotaFiscal.cs
namespace FaturamentoService.Models
{
    public enum StatusNotaFiscal
    {
        Aberta,
        Fechada
    }

    public class NotaFiscal
    {
        public int Id { get; set; }
        public int Numero { get; set; }
        public StatusNotaFiscal Status { get; set; } = StatusNotaFiscal.Aberta;
        public DateTime DataCriacao { get; set; } = DateTime.UtcNow;

        public List<NotaFiscalItem> Itens { get; set; } = new();
    }
}

File: frontend/src/app/models/nota-fiscal.ts
export enum StatusNotaFiscal {
  Aberta = 0,
  Fechada = 1
}

export interface NotaFiscalItem {
  id?: number;
  produtoId: number;
  quantidade: number;
}

export interface NotaFiscal {
  id: number;
  numero: number;
  status: StatusNotaFiscal;
  dataCriacao: string;
  itens: NotaFiscalItem[];
}

export interface NovaNotaFiscal {
  itens: { produtoId: number; quantidade: number }[];
}